docs/basic-features/environment-variables.md

```diff
@@ -0,0 +1,108 @@
+---
+description: Learn to add and access environment variables in your Next.js application.
+---
+
+# Environment Variables
+
+> This document is for Next.js versions 9.4 and up. If you’re using an older version of Next.js, refer to [Environment Variables in next.config.js](/docs/api-reference/next.config.js/environment-variables.md).
+
+Next.js comes with built-in support for environment variables, which allows you to do the following:
+
+- [Inline variables starting with `NEXT_PUBLIC_`](#inlined-environment-variables)
+- [Use `.env` to add custom environment variables](#exposing-environment-variables)
+
+## Inlined Environment Variables
+
+Next.js will inline any environment variable that starts with `NEXT_PUBLIC_` using Webpack [DefinePlugin](https://webpack.js.org/plugins/define-plugin/). Inlining means replacing the variable with the value. For example, the following page:
+
+```jsx
+export default function Page() {
+  return <h1>The public value is: {process.env.NEXT_PUBLIC_EXAMPLE_KEY}</h1>
+}
+```
+
+Will end up being:
+
+```jsx
+export default function Page() {
+  return <h1>The public value is: {'my-value'}</h1>
+}
+```
+
+Next.js replaced `process.env.NEXT_PUBLIC_EXAMPLE_KEY` with its value, that in this case is `'my-value'`.
+
+You can use the shell or any other tool that runs before the [Next.js CLI](/api-reference/cli) to add environment variables. For example, using the bash:
+
+```bash
+NEXT_PUBLIC_EXAMPLE_KEY=my-value npx next dev
+```
+
+And using [cross-env](https://github.com/kentcdodds/cross-env):
+
+```bash
+npx cross-env NEXT_PUBLIC_EXAMPLE_KEY=my-value next dev
+```
+
+### Caveats
+
+- Trying to destructure `process.env` variables won't work due to the nature of webpack [DefinePlugin](https://webpack.js.org/plugins/define-plugin/).
+- When dealing with secrets, it's better to [expose the variables using `.env`](#exposing-environment-variables), to avoid exposing secrets in your build output
```

Comment:
```suggestion
- To avoid exposing secrets, do not use the `NEXT_PUBLIC_` for them. Instead, [expose the variables using `.env`](#exposing-environment-variables).
```